app/templates/components/public/ticket-list.hbs

```diff
@@ -125,9 +125,10 @@
                       placeholder=(t 'Promotional Code')
                       readonly=promotionalCodeApplied
                       key-up=(action 'handleKeyPress')
+                      id='promotionalCode'
               }}
               {{#unless promotionalCodeApplied}}
-                <div role="button" class="ui icon button" {{action 'applyPromotionalCode'}}>
+                <div role="button" class="ui icon {{if promotionalCode '' 'disabled'}} button" {{action 'applyPromotionalCode'}}>
```

Comment:
@maze-runnar Is this really needed ? Validation should be taking care for empty input .

Reply:
but it was still calling the server and reloading the page.

Reply:
Are you sure ?